storage-shared-benchmarking/src/main/java/com/google/cloud/storage/benchmarking/StorageSharedBenchmarkingCli.java

```diff
@@ -108,35 +108,50 @@ private void runWorkload1() {
     StorageOptions retryStorageOptions =
         StorageOptions.newBuilder().setProjectId(project).setRetrySettings(retrySettings).build();
     Storage storageClient = retryStorageOptions.getService();
-    runW1R3(storageClient);
+    try {
+      runW1R3(storageClient);
+    } catch (Exception e) {
+      System.err.println("Failed to run workload 1: " + e.getMessage());
+    }
   }
 
   private void runWorkload4() {
     RetrySettings retrySettings = StorageOptions.getDefaultRetrySettings().toBuilder().build();
     StorageOptions retryStorageOptions =
         StorageOptions.grpc().setRetrySettings(retrySettings).setAttemptDirectPath(true).build();
     Storage storageClient = retryStorageOptions.getService();
-    runW1R3(storageClient);
+    try {
+      runW1R3(storageClient);
+    } catch (Exception e) {
+      System.err.println("Failed to run workload 4: " + e.getMessage());
+    }
   }
 
-  private void runW1R3(Storage storageClient) {
+  private void runW1R3(Storage storageClient) throws ExecutionException, InterruptedException {
     Path tempDir =
         tempDirLocation != null
             ? Paths.get(tempDirLocation)
             : Paths.get(System.getProperty("java.io.tmpdir"));
     ListeningExecutorService executorService =
         MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(workers));
-    List<ApiFuture<String>> workloadRuns = new ArrayList<>();
     Range objectSizeRange = Range.of(objectSize);
     for (int i = 0; i < samples; i++) {
       int objectSize = getRandomInt(objectSizeRange.min, objectSizeRange.max);
       PrintWriter pw = new PrintWriter(System.out, true);
-      workloadRuns.add(
-          convert(
+      long startTime = System.currentTimeMillis();
+      long endTime = startTime + (warmup * 1000);
+      // Run Warmup
+      while (System.currentTimeMillis() < endTime) {
```

Comment:
It seems perhaps a little confusing to have this as just a few lines within W1R3 rather than as a separate method. Not a java expert here but maybe it's better to factor this out for ease of readability and/or sharing with other workloads?